src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -940,31 +964,22 @@ private void GetLambdaId(SyntaxNode syntax, ClosureKind closureKind, int closure
 
             Debug.Assert(!isLambdaBody || LambdaUtilities.IsLambdaBody(lambdaOrLambdaBodySyntax));
 
-            // determine lambda ordinal and calculate syntax offset:
-            lambdaOrdinal = _lambdaDebugInfoBuilder.Count;
-            int syntaxOffset = _topLevelMethod.CalculateLocalSyntaxOffset(lambdaOrLambdaBodySyntax.SpanStart, lambdaOrLambdaBodySyntax.SyntaxTree);
-            _lambdaDebugInfoBuilder.Add(new LambdaDebugInfo(syntaxOffset, closureOrdinal));
+            // determine lambda ordinal and calculate syntax offset
 
-            int previousLambdaOrdinal;
-            if (slotAllocatorOpt != null &&
-                slotAllocatorOpt.TryGetPreviousLambda(lambdaOrLambdaBodySyntax, isLambdaBody, out previousLambdaOrdinal))
+            DebugId lambdaId;
+            DebugId previousLambdaId;
+            if (slotAllocatorOpt != null && slotAllocatorOpt.TryGetPreviousLambda(lambdaOrLambdaBodySyntax, isLambdaBody, out previousLambdaId))
             {
-                topLevelMethodId = slotAllocatorOpt.PreviousMethodId;
-                lambdaOrdinal = previousLambdaOrdinal;
+                lambdaId = previousLambdaId;
             }
```

Comment:
Perhaps remove 'previousLambdaId' and pass 'lambdaId' instead. Same comment for VB.